tests/test_engine_responses_tool_calls.py

```
@@ -72,6 +73,53 @@ async def get_response(self, *_args, **_kwargs) -> str:
 
+@pytest.mark.asyncio
+async def test_finish_response_tool_call_is_not_persisted_or_emitted() -> None:
+    class _Handler:
+        def __init__(self) -> None:
+            self._tool_call = {
+                "name": "finish_response",
+                "arguments": "{}",
+                "call_id": "call_finish_response",
+            }
+
+        def get_and_clear_last_tool_call(self):
```

Ruff (0.15.11)

[warning] Missing return type annotation for private function `get_and_clear_last_tool_call`

(ANN202)